Ray.jl/src/function_manager.jl

```diff
@@ -95,9 +95,9 @@ function wait_for_function(fm::FunctionManager, fd::ray_jll.JuliaFunctionDescrip
                            pollint_s=0.01, timeout_s=10)
     key = function_key(fd, job_id)
     status = timedwait(timeout_s; pollint=pollint_s) do
-        # timeout the Exists query to the same timeout we use here so we don't
-        # deadlock.
-        ray_jll.Exists(fm.gcs_client, FUNCTION_MANAGER_NAMESPACE, key, timeout_s)
+        # Set Exists to not block by using a negative timeout. If we set a non-negative
+        # timeout then this function can throw "Deadline Exceeded" exceptions.
+        ray_jll.Exists(fm.gcs_client, FUNCTION_MANAGER_NAMESPACE, key, -1)
```

Comment:
Alternatively we could update the `ray_jll.Exists` function to return `false` when we hit time timeout. Something like:
```cpp
if (!status.ok()) {
    std::string status_str = status.ToString();

    // Key doesn't exist before we hit the timeout
    if (timeout_ms >= 0 && status_str.find("Deadline Exceeded") != std::string::npos) {
        return false;
    }
    throw std::runtime_error(status_str);
}
```
This approach means we'd no longer need the `timedwait`